tools/internal_ci/helper_scripts/prepare_build_linux_rc

```diff
@@ -18,12 +18,23 @@
 # Need to increase open files limit for c tests
 ulimit -n 32768
 
-# Move docker's storage location to scratch disk so we don't run out of space.
-echo 'DOCKER_OPTS="${DOCKER_OPTS} --graph=/tmpfs/docker"' | sudo tee --append /etc/default/docker
-# Use container registry mirror for pulling docker images (should make downloads faster)
-# See https://cloud.google.com/container-registry/docs/using-dockerhub-mirroring
-echo 'DOCKER_OPTS="${DOCKER_OPTS} --registry-mirror=https://mirror.gcr.io"' | sudo tee --append /etc/default/docker
-sudo service docker restart
+# 1. Move docker's storage location to scratch disk so we don't run out of space.
+# 2. Use container registry mirror for pulling docker images (should make downloads faster)
+#    See https://cloud.google.com/container-registry/docs/using-dockerhub-mirroring
+GRPC_DOCKER_OPTS="--data-root=/tmpfs/docker --registry-mirror=https://mirror.gcr.io"
+
+# Update global docker configuration to have GRPC_DOCKER_OPTS
+P1NAME=$(ps -p 1 -o comm=)
+if [[ $P1NAME == *"systemd"* ]];  then
+  # https://stackoverflow.com/questions/27763340
+  sudo sed -i -E "s/ExecStart=(.*)\/\//EnvironmentFile=\/etc\/default\/docker\nExecStart=\nExecStart=\1\/\/ \$DOCKER_OPTS/" /lib/systemd/system/docker.service
```

Comment:
Nit: Maybe use "|" as the separator instead of "/" so you don't have to use the escape character?

Reply:
What's the purpose of the first "ExecStart=\n"? I'm not an expert in systemd, but I'd expect this to be overridden by the next line.

Reply:
Oh different separators are allowed to use! Thanks for the suggestion.

Redundant `ExecStart=` doesn't seem straightforward but they're saying it's required to clear the existing one to be able to set it again.